packages/repo-tools/src/commands/openapi/test.ts

```diff
@@ -0,0 +1,148 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { join } from 'path';
+import chalk from 'chalk';
+import { runner } from './runner';
+import { YAML_SCHEMA_PATH } from './constants';
+import { promisify } from 'util';
+import { exec as execCb } from 'child_process';
+import { paths as cliPaths } from '../../lib/paths';
+
+const exec = promisify(execCb);
+
+async function test(
+  directoryPath: string,
+  { port }: { port: number },
+  options?: { update?: boolean },
+) {
+  const openapiPath = join(directoryPath, YAML_SCHEMA_PATH);
+  if (!(await fs.pathExists(openapiPath))) {
+    return;
+  }
+  const opticConfigFilePath = join(directoryPath, 'optic.yml');
+  if (!(await fs.pathExists(opticConfigFilePath))) {
+    if (process.env.CI) {
+      throw new Error(
+        `You must run 'yarn backstage-repo-tools schema openapi test' locally and commit the generated files.`,
+      );
+    }
+    console.error(
+      `No 'optic.yml' file found in '${directoryPath}'. Creating one now, please commit it.`,
+    );
+    await fs.writeFile(
+      opticConfigFilePath,
+      `ruleset:
+  - breaking-changes
+capture:
+  ${YAML_SCHEMA_PATH}:
+    # 🔧 Runnable example with simple get requests.
+    # Run with "PORT=3000 optic capture ${YAML_SCHEMA_PATH} --update interactive" in '${directoryPath}'
+    # You can change the server and the 'requests' section to experiment
+    server:
+      # This will not be used by 'backstage-repo-tools schema openapi test', but may be useful for interactive updates.
+      url: http://localhost:3000
+    requests:
+      # ℹ️ Requests should be sent to the Optic proxy, the address of which is injected into 'run.command's env as OPTIC_PROXY (or the value of 'run.proxy_variable').
+      run:
+        # 🔧 Specify a command that will generate traffic
+        command: yarn backstage-cli package test
+`,
+    );
+    if (
+      (await cliPaths.resolveTargetRoot('node_modules/.bin/prettier')) &&
+      options?.update
+    ) {
+      await exec(`yarn prettier --write ${opticConfigFilePath}`);
+    }
+  }
+
+  try {
+    await exec(
+      `yarn run -T optic capture ${YAML_SCHEMA_PATH} --server-override "http://localhost:${port}" ${
+        options?.update ? '--update' : ''
+      }`,
+      {
+        cwd: directoryPath,
+        env: {
+          ...process.env,
+          PORT: `${port}`,
+          CI: '1',
+          BACKSTAGE_TEST_DISABLE_DOCKER: '1',
+        },
+      },
+    );
+  } catch (err) {
+    // Optic outputs the actual results to stdout, but that will not be added to the message by default.
+    err.message = err.stderr + err.stdout;
+    if (
+      !(err.message as string).includes(
+        'Error: No requests were captured by the Optic proxy',
```

Comment:
This should be considered an error shouldn't it? If no requests were captured we effectively didn't test anything?

Reply:
I think this is an artifact from a previous iteration. My goal here was to ignore test runs from plugins that hadn't setup the network proxy and optic config file, but it's fair to assume if they've set up the optic config file that they should get input if there are test results. Will update.